erpnext/assets/doctype/asset/depreciation.py

```diff
@@ -132,6 +134,28 @@ def get_depreciation_accounts(asset):
 
 	return fixed_asset_account, accumulated_depreciation_account, depreciation_expense_account
 
+def get_credit_and_debit_accounts(accumulated_depreciation_account, depreciation_expense_account):
+	if is_income_or_expense_account(depreciation_expense_account) == "Expense":
+		credit_account = accumulated_depreciation_account
+		debit_account = depreciation_expense_account
+	else:
+		credit_account = depreciation_expense_account
+		debit_account = accumulated_depreciation_account
+
+	return credit_account, debit_account
+
+def is_income_or_expense_account(account):
+	from frappe.utils.nestedset import get_ancestors_of
+
+	ancestors = [ancestor.split(' - ')[0] for ancestor in get_ancestors_of("Account", account)]
```

Comment:
Why not just check the root type of the account?

Reply:
This feature was introduced because according to Rijul, customers in the European region prefer to use the same root node for both Income and Expense accounts, which is then divided into Income and Expense nodes at the next level. So checking the root type wouldn't be helpful here(although it would obviously have been more efficient).

Reply:
If `root_type` field of selected account is either `expense` or `income` then it is a valid account, right?
Is there a reason to check all the ancestors of the selected account?

Reply:
From what I understood, it sounds like such customers have four nodes at the first level- one for Assets, one for Liabilities, one that's used for both Income and Expenses(let's call this X), and a final dummy node. X is then further split into Income and Expenses. So wouldn't the `root_type` of any income or expense account just be X?